src/sentry/event_manager.py

```diff
@@ -600,7 +602,10 @@ def save(self, project, raw=False):
         if fingerprint:
             hashes = [md5_from_hash(h) for h in get_hashes_from_fingerprint(event, fingerprint)]
```

Comment:
not so much related to this PR, but just a clarification on the snuba stuff, if primary_hash is just the hash of the first element, does that mean that an event with fingerprint `['foo', 'bar']` has the same primary hash as one with a fingerprint of `['foo', 'baz']` and is that the correct matching behavior we want?

Reply:
Stream of consciousness here:

* an event can have multiple hashes/fingerprints: `[a, b, c]`
* hexdigest hashes are calculated in `event_manager.save`
* each one is used in a `GroupHash.objects.get_or_create(project, hash)`
* those are iterated over and the first `GroupHash` that already has a `group_id` is used as the group, else a new group is created
* that `group_id` is associated with *all* of the `GroupHash` values from earlier (representing `[a, b, c]`)

So we use the first hash at write time in Snuba... and at query time (when searching a group) we will always send all hashes associated with the group.

That was my brain dump... if you see any holes now is the time for us to change stuff. :fearful:

Reply:
I guess there is some fuzziness around merge and unmerge and what it means when the `primary_hash` for a given set of events is moved between groups. @tkaemming would probably have a better idea here.

Reply:
Nothing really significant with merging will change here. Unmerging will be a little weird (you'd have to move two `GroupHash` records when previously you only had to move one) but I don't think that's a significant enough problem to worry about based on the low utilization of `checksum` anyway.